execution/execute.go

```diff
@@ -159,6 +167,67 @@ func newExecution(executionInfo *executionInfo) suiteExecutor {
 	return newSimpleExecution(executionInfo, true)
 }
 
+type executionStatus struct {
+	SpecsExecuted int
+	SpecsPassed   int
+	SpecsFailed   int
+	SpecsSkipped  int
+	SceExecuted   int
+	ScePassed     int
+	SceFailed     int
+	SceSkipped    int
+}
+
+func newExecutionStatus() *executionStatus {
+	return &executionStatus{SpecsExecuted: 0, SpecsPassed: 0, SpecsFailed: 0, SpecsSkipped: 0, SceExecuted: 0, ScePassed: 0, SceFailed: 0, SceSkipped: 0}
+}
+
+func (status *executionStatus) getJSON() (string, error) {
+	j, err := json.MarshalIndent(status, "", "\t")
+	if err != nil {
+		return "", err
+	}
+	return string(j), nil
+}
+
+func writeExecutionStatus(executedSpecs int, passedSpecs int, failedSpecs int, skippedSpecs int, executedScenarios int, passedScenarios int, failedScenarios int, skippedScenarios int) {
```

Comment:
Don't need to specify the type for all params if the type is same. It can be `function(a,b,c int)`